core/include/moveit/task_constructor/solvers/cartesian_path.h

```diff
@@ -63,11 +63,11 @@ class CartesianPath : public PlannerInterface
 
 	void init(const moveit::core::RobotModelConstPtr& robot_model) override;
 
-	Result plan(const planning_scene::PlanningSceneConstPtr& from, const planning_scene::PlanningSceneConstPtr& to,
+	MoveItErrorCode plan(const planning_scene::PlanningSceneConstPtr& from, const planning_scene::PlanningSceneConstPtr& to,
```

Comment:
Why did you decide to introduce Result instead of using error codes? Aren't we loosing information by reducing the error code to a boolean?

Reply:
Originally, #525 introduced `tl::expected<bool, std::string>` to return both, a boolean result and an error message. Later, this was changed to `MoveItErrorCode`, which - in ROS2 - supports the error message and an error source. As this extended MoveItErrorCode is not available in ROS1 and the purpose was just to return an additional string message, I introduced an explicit data type that is uniformly usable in ROS1 and ROS2.
If at all, the source information is lost (it was never fed into the error code).

Reply:
I don't consider reverting to MoveItErrorCode and I don't see something missing so far.

Reply:
> If at all, the source information is lost (it was never fed into the error code).

Good point that should be done, otherwise that field is pointless.

I see, have you considered enhancing the error code with a string message in moveit1? The main motivation here was to improve the debug-ability of moveit in general. The source field was introduced to infer the planning pipeline that created the error message when using parallel planning. Without that, I think it is just possible to tell that one of the planners failed with the given error message. Not which one. Alternatively, the string could be enhanced with that information.

Reply:
Yes, I shortly considered augmenting MoveItErrorCode in MoveIt1 as well. However, I decided against, because the error message is redundant as we already have a fixed mapping from the error code to a human-readable message (`MoveItErrorCode::toString()`). Further, the source member was not filled regularly - only in planning adapters. As far as I can see, the planning pipeline name is not yet stored in that field.

Note, that there are _two_ MoveItErrorCode header files (I think one should be removed):
- `moveit_core/utils/include/moveit/utils/moveit_error_code.h` (regularly used)
- `moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/include/moveit/hybrid_planning_manager/moveit_error_code_interface.h` (used only once)